### PR TITLE
Greentrip 2019 06 08

### PR DESCRIPTION
**Pull request by trevor-wu**
fitted modeconfirm to destination confirm
Screen Shot 2019-06-16 at 5 20 48 AM
Made pick destination the sole option on the menu.
Made it impossible to enter details page without having selected a mode first.
Screen Shot 2019-06-16 at 5 23 55 AM
Selected mode, now name of destination appears at top of details page along with its star rating in order to make it easier to reference with the suggestion.
Screen Shot 2019-06-16 at 5 24 09 AM
Suggestion loads upon entering page, along with its star rating. Additional reviews can be accessed by clicking the button.
Screen Shot 2019-06-16 at 5 24 19 AM
If the user chooses to see more reviews, they can click the button and the button disappears after reviews are loaded.

### CODE REVIEW

File: www/css/main.diary.css [hunk 1]
```diff
@@ -46,8 +46,8 @@
 /* ----------- iPhone 5 and 5S ----------- */
 
 /* Portrait and Landscape */
-@media only screen
```
Comment (shankari) [inline www/css/main.diary.css]: @trevor-wu can you please set your editor to not modify whitespace? It messes up the git blame.
As we discussed earlier "Hide whitespace changes" just hides the changes in the review; they still make it through to the git history.

File: www/js/diary/list.js [hunk 2]
```diff
@@ -692,7 +697,7 @@ angular.module('emission.main.diary.list',['ui-leaflet',
     };
 
     /*
-     * Convert the array of {text, value} objects to a {value: text} map so that
```
Comment (shankari) [inline www/js/diary/list.js]: whitespace

File: www/js/diary/list.js [hunk 3]
```diff
@@ -776,7 +782,7 @@ angular.module('emission.main.diary.list',['ui-leaflet',
     $scope.checkTripState = function() {
       window.cordova.plugins.BEMDataCollection.getState().then(function(result) {
         Logger.log("Current trip state" + JSON.stringify(result));
-        if(JSON.stringify(result) ==  "\"STATE_ONGOING_TRIP\"" || 
+        if(JSON.stringify(result) ==  "\"STATE_ONGOING_TRIP\"" ||
```
Comment (shankari) [inline www/js/diary/list.js]: whitespace

File: www/templates/diary/list.html [hunk 4]
```diff
@@ -3,7 +3,7 @@
     </ion-nav-bar>
     <ion-nav-buttons side="right">
         <button class="button button-icon ion-help" ng-click="startWalkthrough()"></button>
-    </ion-nav-buttons>
```
Comment (shankari) [inline www/templates/diary/list.html]: whitespace

**Comment by shankari** [comment]
I also don't see any evidence that this happened

> fitted modeconfirm to destination confirm

The UX has changed visually, but the data still seems to be stored as `mode_confirm`

>  Selected mode, now name of destination appears at top of details page along with its star rating in order to make it easier to reference with the suggestion.

But destination != mode.

**Comment by shankari** [comment]
Ah I see that you tried to change the server code to have the suggestion system server code work off `mode_confirm` instead of `destination_confirm`. That is the wrong change. Labels matter. Please change the front end instead.

I think you are almost there, please:
- merge master -> greentrip in a separate PR
- change the front end to use `destination_confirm` instead of `mode_confirm`
- remove the extraneous whitespaces

and then I can review and submit.

**Comment by trevor-wu** [comment]
Filepath change to destination_confirm works with server, tested
Screen Shot 2019-07-06 at 1 51 10 AM
Screen Shot 2019-07-06 at 1 50 54 AM

File: www/js/diary/list.js [hunk 5]
```diff
@@ -726,13 +731,14 @@ angular.module('emission.main.diary.list',['ui-leaflet',
       $scope.draftMode.label = mode.value;
       Logger.log("in storeMode, after setting mode.value = " + mode.value + ", draftMode = " + JSON.stringify($scope.draftMode));
       var tripToUpdate = $scope.modeTripgj;
-      Timeline.instantSave(MODE_CONFIRM_KEY, $scope.draftMode).then(function () {
+      Timeline.instantSave(DESTINATION_CONFIRM_KEY, $scope.draftMode).then(function () {
```
Comment (shankari) [inline www/js/diary/list.js]: like here. You are not saving the `draftMode` to the server, but the `draftDest`. Renaming is a purely syntactic change, not a semantic change, but it makes it a lot easier for others to read and understand.

**Comment by trevor-wu** [comment]
Testing, change of variables did not affect functionality:
Screen Shot 2019-07-06 at 3 47 45 PM

**Pull request merged**